### PR TITLE
Add tests for BZ 1262029

### PR DESCRIPTION
**Pull request by Ichimonji10**
Test results:

    $ nosetests tests/foreman/api/test_environment.py:MissingAttrTestCase
    S
    ----------------------------------------------------------------------
    Ran 1 test in 0.548s

    OK (SKIP=1)

Test result without `skip_if_bug_open` decorator:

    $ nosetests tests/foreman/api/test_environment.py:MissingAttrTestCase
    EE
    ======================================================================
    ERROR: @Test: Update an environment. Inspect the server's response.
    ----------------------------------------------------------------------
    Traceback (most recent call last):
    File "/home/ichimonji10/code/robottelo/tests/foreman/api/test_environment.py", line 94, in test_location
        _get_entity_ids('location', self.env_attrs)
    File "/home/ichimonji10/code/nailgun/nailgun/entity_mixins.py", line 280, in _get_entity_ids
        attrs.keys()
    MissingValueError: Cannot find a value for the "location" field. Searched for keys named ('location_ids', 'location', 'locations'), but available keys are [u'hosts_count', u'name', u'created_at', u'updated_at', u'id', u'hostgroups_count', u'katello_id'].

    ======================================================================
    ERROR: @Test: Update an environment. Inspect the server's response.
    ----------------------------------------------------------------------
    Traceback (most recent call last):
    File "/home/ichimonji10/code/robottelo/tests/foreman/api/test_environment.py", line 104, in test_organization
        _get_entity_ids('organization', self.env_attrs)
    File "/home/ichimonji10/code/nailgun/nailgun/entity_mixins.py", line 280, in _get_entity_ids
        attrs.keys()
    MissingValueError: Cannot find a value for the "organization" field. Searched for keys named ('organization_ids', 'organization', 'organizations'), but available keys are [u'hosts_count', u'name', u'created_at', u'updated_at', u'id', u'hostgroups_count', u'katello_id'].

    ----------------------------------------------------------------------
    Ran 2 tests in 3.920s

    FAILED (errors=2)

### CODE REVIEW

**Comment by elyezer** [comment]
ACK

File: tests/foreman/api/test_environment.py [hunk 1]
```diff
@@ -8,8 +8,9 @@
 from ddt import ddt
 from fauxfactory import gen_string
 from nailgun import entities
+from nailgun.entity_mixins import _get_entity_ids
```
Comment (sthirugn) [inline tests/foreman/api/test_environment.py]: I am not comfortable calling a function prefixed with _ If this function is to be usable by other modules, please make it public. Prefixed with _ has a reason and I dont want us to change it. Non-blocker for this PR though

**Comment by sthirugn** [comment]
ACK with a non-blocker comment

File: tests/foreman/api/test_environment.py (hunk 1 above)
Comment (Ichimonji10) [inline tests/foreman/api/test_environment.py]: Same response as https://github.com/SatelliteQE/robottelo/pull/2757#discussion_r39182928

**Pull request merged**